src/Agent.Worker/WorkerCommandManager.cs

```diff
@@ -107,6 +109,13 @@ public bool TryProcessCommand(IExecutionContext context, string input)
                     {
                         extension.ProcessCommand(context, command);
                     }
+                    catch (SocketException ex)
+                    {
+                        #pragma warning disable CA2000 // Dispose objects before losing scope
```

Comment:
Do we need to suppress CA2000 check?

Reply:
Yes. Otherwise compiler returns error "error CA2000: Call System.IDisposable.Dispose on object created by 'WorkerUtilities.GetVssConnection(context)' before all references to it are out of scope.".
I found that suppression of error CA2000 is used in other places in agent code, so I used this also to mitigate this issue.

Reply:
I see thanks - could it make sense to use attribute for a method then instead of pragma directive?